lib/std/math.zig

```diff
@@ -596,6 +596,8 @@ test "shr" {
 /// Rotates right. Only unsigned values can be rotated.  Negative shift
 /// values result in shift modulo the bit count.
 pub fn rotr(comptime T: type, x: T, r: anytype) T {
+    if (T == u0) @compileError("cannot rotate u0");
```

Comment:
>  I thought it might be a good idea to make passing u0 a compile error.

IMO it's reasonable for this just to return `0`

Also, the Vector case needs to be handled I think

Reply:
Thanks, will do.